### PR TITLE
fix crash in serveXdsServer when failed to listen

### PR DESCRIPTION
**Pull request by muyuan0**
The serveXdsServer will crash if the net.Listen failed, because of:
1. the log call uses an odd number of arguments in argument pair
2. the code continues to execute and will access the nil listener.

This PR fixes the behavior.

### CODE REVIEW

**Comment by Xunzhuo** [comment]
Hey @muyuan0 Thanks for the works.

Just one suggestion. Before getting started, could you raise an issue first, in case maintainers can have a good review to understand what we are solving now?

**Comment by muyuan0** [comment]
> Hey @muyuan0 Thanks for the works.
> 
> Just one suggestion. Before getting started, could you raise an issue first, in case maintainers can have a good review to understand what we are solving now?

Thanks. I will raise an issue next time.

**Comment by muyuan0** [comment]
As for this case, I think the test case is already clear enough to show my intention. That's why I didn't open an issue first.

**Comment by zirain** [comment]
an issue or PR description is needed.

**Comment by muyuan0** [comment]
> an issue or PR description is needed.

Updated the PR description.

File: tools/make/golang.mk [hunk 1]
```diff
@@ -41,12 +41,13 @@ go.build.multiarch: $(foreach p,$(PLATFORMS),$(addprefix go.build., $(addprefix
 
 .PHONY: go.test.unit
 go.test.unit: ## Run go unit tests
-	go test ./...
+	# The gcflags here ensure gomonkey always work, see https://github.com/agiledragon/gomonkey#notes
+	go test -gcflags=all=-l ./...
```
Comment (arkodg) [inline tools/make/golang.mk:45]: im reluctant to add this,
am I'm fine with the code w/o the test (with monkey patch to mock it)

Comment (muyuan0) [reply tools/make/golang.mk:45]: The gomonkey can work without the no-inline flag in this case.
Could you tell me the reason why the `gcflags` is not welcome here? I always add `gcflags="all=-N -l"` with `go build` to make the debugger work usefully so I am comfortable with new gcflags...

**Review by arkodg — APPROVED** [review]
LGTM, thanks for fixing this and also adding the test case to validate it !

**Comment by muyuan0** [comment]
Sorry for the rebase by accident!

**Review by AliceProxy — APPROVED** [review]
lgtm, thanks for fixing this

**Review by arkodg — APPROVED** [review]

**Pull request merged**